### PR TITLE
Updated basics/nested-routing to v4.

### PR DESCRIPTION
**Pull request by cgubbin**
Changed header attachment in test to use `insert_header`.

### CODE REVIEW

**Review by robjtede — CHANGES_REQUESTED** [review]

File: basics/nested-routing/Cargo.toml [hunk 1]
```diff
@@ -5,14 +5,14 @@ authors = ["dowwie <dkcdkg@gmail.com>"]
...
+env_logger = "0.9"
 futures = "0.3.1"
 serde = { version = "^1.0", features = ["derive"] }
 serde_json = "1.0.39"
-time = "0.1.42"
+time = "0.3.7"
 
 [dev-dependencies]
-actix-rt = "1"
+actix-rt = "2.6.0"
```
[the remark this answers is not in the document]
Comment (cgubbin) [reply basics/nested-routing/Cargo.toml]: I fixed this. Thank you.

File: basics/nested-routing/Cargo.toml [hunk 2]
```diff
@@ -5,14 +5,14 @@ authors = ["dowwie <dkcdkg@gmail.com>"]
 edition = "2018"
 
 [dependencies]
-actix-web = { version = "3", features = ["openssl"] }
-actix-service = "1.0.0"
-bytes = "0.5.3"
-env_logger = "0.8"
+actix-web = { version = "4.0.0-beta.21", features = ["openssl"] }
+actix-service = "2.0.2"
```
[the remark this answers is not in the document]
Comment (cgubbin) [reply basics/nested-routing/Cargo.toml]: I removed actix-service and imported the from actix-web::dev.

File: basics/nested-routing/Cargo.toml [hunk 3]
```diff
@@ -5,14 +5,14 @@ authors = ["dowwie <dkcdkg@gmail.com>"]
...
+bytes = "1.1.0"
```
[the remark this answers is not in the document]
Comment (cgubbin) [reply basics/nested-routing/Cargo.toml]: The `bytes` crate was actually an unused dependancy. I removed it. I am sorry this one was so messy, it was my first ever pull request! I have a much better idea what I'm doing now.

**Review by robjtede — APPROVED** [review]
Nice one. Thanks.

**Pull request merged**